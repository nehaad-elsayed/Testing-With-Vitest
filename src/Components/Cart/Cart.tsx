


export default function Cart({items}: {items: string[]}) {
  return (
    <div className="cart">
      <h2>Cart</h2>
      {items.length === 0 ? <p>No items in your cart</p> : <p>You have {items.length} items in your cart</p>}
    </div>
  );
}
