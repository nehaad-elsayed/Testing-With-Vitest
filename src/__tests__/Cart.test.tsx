import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Cart from "../Components/Cart/Cart";


test("snapshot with nothing in the cart", () => {
    const {asFragment} = render(<Cart items={[]}/>);
    expect(asFragment()).toMatchSnapshot();
});

test("snapshot with one item in the cart", () => {
    const {asFragment} = render(<Cart items={["Pizza"]}/>);
    expect(asFragment()).toMatchSnapshot();
});

test("snapshot with multiple items in the cart", () => {
    const {asFragment} = render(<Cart items={["Pizza", "Pasta", "Burger"]}/>);
    expect(asFragment()).toMatchSnapshot();
});



