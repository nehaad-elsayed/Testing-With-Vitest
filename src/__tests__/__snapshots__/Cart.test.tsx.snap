// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot with multiple items in the cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <p>
      You have 3 items in your cart
    </p>
  </div>
</DocumentFragment>
`;

exports[`snapshot with nothing in the cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <p>
      No items in your cart
    </p>
  </div>
</DocumentFragment>
`;

exports[`snapshot with one item in the cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <p>
      You have 1 items in your cart
    </p>
  </div>
</DocumentFragment>
`;
